packages/tools/client-debugger/client-debugger/src/test/ClientDebugger.test.ts

```diff
@@ -119,6 +119,32 @@ describe("ClientDebugger unit tests", () => {
 		expect(audienceRemoved).to.be.true;
 	});
 
+	it("Validate container state contents are as expected ", async () => {
+		const clientDebugger = initializeDebugger(containerId, container!);
+
+		// verify container state change in container is reflecting in client debugger
+		container!.connect();
+
+		expect(clientDebugger?.getContainerConnectionLog().length).to.equal(1);
+		expect(clientDebugger?.getContainerConnectionLog()[0].newState).to.equal("connected");
+
+		await container!.attach({ url: "test-url" });
+		expect(clientDebugger?.getContainerConnectionLog().length).to.equal(2);
+		expect(clientDebugger?.getContainerConnectionLog()[1].newState).to.equal("attached");
+
+		container!.disconnect();
+		expect(clientDebugger?.getContainerConnectionLog().length).to.equal(3);
+		expect(clientDebugger?.getContainerConnectionLog()[2].newState).to.equal("disconnected");
+
+		container!.close();
+		expect(clientDebugger?.getContainerConnectionLog().length).to.equal(4);
+		expect(clientDebugger?.getContainerConnectionLog()[3].newState).to.equal("closed");
+
+		container!.dispose?.();
```

Comment:
The dispose method is always present on our Mock Container implementation - do we need the `?.` in the call?

Reply:
Yes. The `IContainer` define dispose function as `dispose?(error?: ICriticalContainerError): void;`